conda_build/build.py

```diff
@@ -438,6 +454,99 @@ def create_info_files(m, files, config, prefix):
                   config.timeout)
 
 
+def get_short_path(m, target_file):
+    entry_point_script_names = get_entry_point_script_names(m.get_value('build/entry_points'))
+    if is_noarch_python(m):
+        if target_file.find("site-packages") > 0:
+            return target_file[target_file.find("site-packages"):]
+        elif target_file.startswith("bin") and (target_file not in entry_point_script_names):
+            return target_file.replace("bin", "python-scripts")
+        elif target_file.startswith("Scripts") and (target_file not in entry_point_script_names):
+            return target_file.replace("Scripts", "python-scripts")
+    elif m.get_value('build/noarch_python'):
+        return None
+    else:
+        return target_file
+
+
+def sha256_checksum(filename, buffersize=65536):
+    sha256 = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        for block in iter(lambda: f.read(buffersize), b''):
+            sha256.update(block)
+    return sha256.hexdigest()
+
+
+def has_prefix(short_path, files_with_prefix):
+    for prefix, mode, filename in files_with_prefix:
+        if short_path == filename:
+            return prefix, mode
+    return None, None
+
+
+def is_no_link(no_link, short_path):
+    if no_link is not None and short_path in no_link:
+        return True
+
+
+def get_inode_paths(files, target_short_path, prefix):
+    ensure_list(files)
+    target_short_path_inode = os.stat(join(prefix, target_short_path)).st_ino
+    hardlinked_files = [sp for sp in files
+                        if os.stat(join(prefix, sp)).st_ino == target_short_path_inode]
```

Comment:
Oh right, I forgot about `lstat()`. 

Though I don't think `samestat` would be good here. The docs indicate that it is available on Unix. Opposed to `os.stat(path).st_ino` which is available on Unix and Windows